include/arch/arm/arch/32/mode/machine/fpu.h

```diff
@@ -88,8 +88,9 @@ static void setEnFPEXC(void)
     VMSR(FPEXC, fpexc);
 }
```

Comment:
The commit message describes: "Use the task flag for the host and always enable FPU for the guest". Seems like there's a policy for whether the guest has access to FPU and on x86, TCB's have a second flag defined for guest FPU access?

Reply:
How I implemented it, with x86 virtualisation guests will always have the FPU enabled, so no extra flag is needed.

Although that is hard-coding policy in the kernel, with the reality of SIMD being used everywhere on x86 by both user space and more and more the Linux kernel itself too, mostly for non-FPU SIMD operations, I didn't see the added value of introducing a specific flag for x86 guest FPU access.

I could add a special flags just for x86 guests, but I don't think anyone will disable SIMD for guests in practice, so it would just make things more complicated for no gain.

I think it would be much better to give x86 virtual guests their own TCB like Arm has, then it will have its own flag automatically.

Reply:
>  but I don't think anyone will disable SIMD for guests in practice, so it would just make things more complicated for no gain.

I agree. It is policy, but it doesn't really make sense to not have FPU on for x86 VCPUs. We can also still add another flag later if we want to. The current implementation has most of what is needed for that, so it'd be a much smaller step than this one with not much verification impact. At least I would vote for merging this while we discuss more.

The verification PRs for FPU are large and are blocking other merges at the moment, so it'd be good to get this one done. If we're all in agreement that the mechanism as such are fine we can still fine tune in additional PRs.